src/main/java/mate/academy/internetshop/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.internetshop.service.impl;
+
+import java.util.List;
+import mate.academy.internetshop.dao.ShoppingCartDao;
+import mate.academy.internetshop.lib.Inject;
+import mate.academy.internetshop.lib.Service;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.model.ShoppingCart;
+import mate.academy.internetshop.service.ShoppingCartService;
+import mate.academy.internetshop.service.UserService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Inject
+    private UserService userService;
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().add(product);
+        shoppingCartDao.update(shoppingCart);
+        return shoppingCart;
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        if (shoppingCart.getProducts().remove(product)) {
+            shoppingCartDao.update(shoppingCart);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCart.getProducts().clear();
+        shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart getByUserId(Long userId) {
+        return shoppingCartDao.getAll().stream()
+                .filter(shoppingCart -> shoppingCart.getUser().getUserId().equals(userId))
+                .findFirst()
+                .orElseGet(() -> shoppingCartDao.create(new ShoppingCart(userService.get(userId))));
```

Comment:
```suggestion
                .findFirst();
```
якщо id невірний не коректно створювати нову корзину